libvast/src/format/simdjson.cpp

```diff
@@ -0,0 +1,449 @@
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#include "vast/format/simdjson.hpp"
+
+#include "vast/concept/parseable/vast/address.hpp"
+#include "vast/concept/parseable/vast/subnet.hpp"
+#include "vast/concept/parseable/vast/time.hpp"
+#include "vast/concept/parseable/vast/json.hpp"
+#include "vast/concept/printable/to_string.hpp"
+#include "vast/concept/printable/vast/data.hpp"
+#include "vast/concept/printable/vast/json.hpp"
+#include "vast/data.hpp"
+#include "vast/logger.hpp"
+#include "vast/policy/include_field_names.hpp"
+#include "vast/table_slice.hpp"
+#include "vast/table_slice_builder.hpp"
+#include "vast/type.hpp"
+#include "vast/view.hpp"
+
+#include <caf/detail/pretty_type_name.hpp>
+#include <caf/expected.hpp>
+#include <caf/none.hpp>
+
+namespace vast::format::simdjson {
+namespace {
+
+template <typename JsonType>
+struct adjusted_json_param {
+  using type = std::conditional_t<
+    std::is_arithmetic_v<JsonType> || std::is_same_v<JsonType, std::string_view>,
+    JsonType, const JsonType&>;
+};
+
+template <typename JsonType>
+using adjusted_json_param_t = typename adjusted_json_param<JsonType>::type;
+
+template <typename T>
+caf::expected<data>
+convert_from_impl(adjusted_json_param_t<T> v, const type& t);
+
+caf::expected<data> convert(const ::simdjson::dom::element& e, const type& t);
+
+template <typename JsonType, int ConcreteTypeIndex>
+struct parser_traits {
+  using from_type = JsonType;
+  using to_meta_type = caf::detail::tl_at_t<concrete_types, ConcreteTypeIndex>;
+  using to_type = type_to_data<to_meta_type>;
+
+  static constexpr auto can_be_parsed
+    = std::is_same_v<from_type, std::string_view> && has_parser_v<to_type>;
+};
+
+/// A default implementation ot conversion from JSON type
+/// (known as one of simdjson element_type) to an internal data type.
+template <typename JsonType, int ConcreteTypeIndex>
+caf::expected<data>
+type_biassed_convert_impl(adjusted_json_param_t<JsonType> j, const type&) {
+  using ptraits = parser_traits<JsonType, ConcreteTypeIndex>;
+
+  if constexpr (ptraits::can_be_parsed) {
+    using value_type = typename ptraits::to_type;
+    value_type x;
+    if (!make_parser<value_type>{}(std::string{j}, x))
+      return make_error(ec::parse_error, "unable to parse",
+                        caf::detail::pretty_type_name(typeid(value_type)), ":",
+                        std::string{j});
+
+    return x;
+  } else {
+    static_cast<void>(j);
+    // ngrodzitski: Once fmt is available message can be improved.
+    // as simdjson has a `ostream<<element_type` which makes it
+    // "serializable" to fmt.
```

Comment:
If you include `vast/concept/printable/stream.hpp` you can have it already.

Reply:
Greate, thnx